src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.GenericDao;
+import core.basesyntax.factory.AbstractFactory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws IllegalAccessException,
+            InvocationTargetException, InstantiationException, NoSuchMethodException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] fields = clazz.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(GenericDao.class)) {
+                    field.set(instance, AbstractFactory.getNecessaryInstance(clazz));
+                } else {
+                    throw new RuntimeException("Unknown field type!!!");
+                }
```

Comment:
on lines 19-23 let's do the following

1) check if the field's type is BetDao and there is `@Dao` in BetDaoImpl - then set field
2) else - check if the field's type is UserDao and there is `@Dao` in UserDaoImpl - then set field
3) else - throw your custom Exception

(lines 17-18 are ok)